src/test/test-dirent-util.c

```diff
@@ -0,0 +1,212 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <dirent.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <sys/stat.h>
+
+#include "alloc-util.h"
+#include "dirent-util.h"
+#include "fs-util.h"
+#include "mkdir.h"
+#include "path-util.h"
+#include "rm-rf.h"
+#include "stat-util.h"
+#include "string-util.h"
+#include "tmpfile-util.h"
+#include "tests.h"
+
+TEST (test_dirent_ensure_type) {
+        int r;
+        static struct dirent de = {
+                .d_type = DT_UNKNOWN,
+                .d_name = "test",
+        };
+
+        assert_se(de.d_type == DT_UNKNOWN);
+        int dir_fd;
+
+        dir_fd = 0;
+        dirent_ensure_type(dir_fd, &de);
+
+        /* Test when d_name is "." or ".." */
+        strcpy(de.d_name, ".");
+        r = dirent_ensure_type(dir_fd, &de);
+        assert_se(r == 0);
+        assert_se(de.d_type == DT_DIR);
+
+        strcpy(de.d_name, "..");
+        r = dirent_ensure_type(dir_fd, &de);
+        assert_se(r == 0);
+        assert_se(de.d_type == DT_DIR);
+}
+
+TEST (test_dirent_is_file) {
+        _cleanup_(rm_rf_physical_and_freep) char *t = NULL;
+        assert_se(mkdtemp_malloc(NULL, &t) >= 0);
```

Comment:
move this down after the declarations, and move the other declarations together

Reply:
Thank you very much @bluca . Done.